diff --git a/master/buildbot/changes/gitpoller.py b/master/buildbot/changes/gitpoller.py
index a5029de..7b8b6aa 100644
--- a/master/buildbot/changes/gitpoller.py
+++ b/master/buildbot/changes/gitpoller.py
@@ -242,11 +242,12 @@ def _process_changes(self, newRev, branch):
 
         lastRev = self.lastRev.get(branch)
         self.lastRev[branch] = newRev
-        if not lastRev:
-            return
 
         # get the change list
-        revListArgs = [r'--format=%H', '%s..%s' % (lastRev, newRev), '--']
+        if lastRev:
+            revListArgs = [r'--format=%H', '%s..%s' % (lastRev, newRev), '--']
+        else:
+            revListArgs = [r'--format=%H', '%s^..%s' % (newRev, newRev), '--']
         self.changeCount = 0
         results = yield self._dovccmd('log', revListArgs, path=self.workdir)
